nms/app/packages/magmalte/app/state/feg/EquipmentState.js

```diff
@@ -136,3 +233,50 @@ async function getActiveFegGatewayId(
     return '';
   }
 }
+
+type FetchProps = {
+  networkId: network_id,
+  id?: gateway_id,
+  enqueueSnackbar?: (
+    msg: string,
+    cfg: EnqueueSnackbarOptions,
+  ) => ?(string | number),
+};
+
+/**
+ * Fetches and returns the list of gateways under the federation network or
+ * the specific gateway if the id is provided.
+ *
+ * @param {network_id} networkId: Id of the federation network.
+ * @param {gateway_id} id id of the federation gateway.
+ * @param {(msg, cfg,) => ?(string | number),} enqueueSnackbar Snackbar to display error
+ * @returns returns a list of the federation gateways or a federation gateway with the given id.
+ */
+export async function FetchFegGateways(props: FetchProps) {
+  const {networkId, id, enqueueSnackbar} = props;
+  if (id !== undefined && id !== null) {
+    try {
+      const gateway = await MagmaV1API.getFegByNetworkIdGatewaysByGatewayId({
+        networkId: networkId,
+        gatewayId: id,
+      });
+      if (gateway) {
+        return {[id]: gateway};
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching gateway information', {
+        variant: 'error',
+      });
```

Comment:
Either the whole function body should be wrapped in the same `try..catch`, or these two error messages should be different and provide some additional context. Right now it's just redundant.